scene/animation/animation_mixer.cpp

```diff
@@ -399,12 +399,18 @@ void AnimationMixer::get_animation_list(List<StringName> *p_animations) const {
 	}
 }
 
-Ref<Animation> AnimationMixer::get_animation(const StringName &p_name) const {
-	ERR_FAIL_COND_V_MSG(!animation_set.has(p_name), Ref<Animation>(), vformat("Animation not found: \"%s\".", p_name));
-	const AnimationData &anim_data = animation_set[p_name];
+const Ref<Animation> &AnimationMixer::get_animation(const StringName &p_name) const {
+	static const Ref<Animation> EMPTY_ANIMATION = Ref<Animation>();
+	AHashMap<StringName, AnimationData>::ConstIterator it = animation_set.find(p_name);
+	ERR_FAIL_COND_V_MSG(!it, EMPTY_ANIMATION, vformat("Animation not found: \"%s\".", p_name));
+	const AnimationData &anim_data = it->value;
 	return anim_data.animation;
 }
 
+Ref<Animation> AnimationMixer::_get_animation(const StringName &p_name) const {
+	return get_animation(p_name);
+}
+
 bool AnimationMixer::has_animation(const StringName &p_name) const {
```

Comment:
You have missed all the places where `has_animation()` is called! Its called even more than `animation_set.has()` so it's worth fixing all the places where it is used.